src/handlers/deeplinks.ts

```diff
@@ -89,14 +89,16 @@ export default async function handleDeeplink(
         break;
       }
       default: {
-        const addressOrENS = urlObj.pathname?.split('/')?.[1] || '';
-        const isValid = await checkIsValidAddressOrDomain(addressOrENS);
-        if (isValid) {
-          return Navigation.handleAction(Routes.SHOWCASE_SHEET, {
-            address: addressOrENS,
-          });
-        } else {
-          Alert.alert('Uh oh! We couldn’t recognize this URL!');
+        const addressOrENS = urlObj.pathname?.split('/')?.[1];
+        if (addressOrENS) {
+          const isValid = await checkIsValidAddressOrDomain(addressOrENS);
+          if (isValid) {
+            return Navigation.handleAction(Routes.SHOWCASE_SHEET, {
+              address: addressOrENS,
+            });
+          } else {
+            Alert.alert('Uh oh! We couldn’t recognize this URL!');
```

Comment:
But this one is gonna show that thing only if it's addressOrENS. I'm not sure how often it could be but if we somehow open empty deeplink - the app is gonna just ignore it. So the user may keep trying to open that link but the app does just nothing which in my opinion is not ideal. Kinda in the always loading state.

Reply:
yeah that's true, it's currently causing https://github.com/rainbow-me/rainbow/issues/2852 where dapps use the empty deeplink. If anything is added to the deeplink is going to go to this check and should validate it as we want to